### PR TITLE
Enable more codecs and demuxers in FFmpeg

### PR DESCRIPTION
**Pull request by JanuszL**
- changes how FFMpeg is built by enabling all demuxers and decoders to handle MPEG container
- adds a WAR to a problem when av_seek_frame with AVSEEK_FLAG_BACKWARD doesn't seek to a keyframe

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It enables more codecs and demuxers in FFmpeg

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
      changes how FFMpeg is built by enabling all demuxers and decoders to handle MPEG container
      adds a WAR to a problem when av_seek_frame with AVSEEK_FLAG_BACKWARD doesn't seek to a keyframe
 - Affected modules and functionalities:
     Dockerfile.deps
     video_loader
 - Key points relevant for the review:
     NA
 - Validation and testing:
     present tests apply
 - Documentation (including examples):
     FFmpeg build guide updated

Relates to https://github.com/NVIDIA/DALI/issues/2643
Required https://github.com/NVIDIA/DALI_extra/pull/51
**JIRA TASK**: *[NA]*

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

File: qa/setup_dali_extra.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 # Fetch test data
 export DALI_EXTRA_PATH=${DALI_EXTRA_PATH:-/opt/dali_extra}
-export DALI_EXTRA_URL=${DALI_EXTRA_URL:-"https://github.com/NVIDIA/DALI_extra.git"}
+export DALI_EXTRA_URL="https://github.com/JanuszL/DALI_extra/"
```
[the remark this answers is not in the document]
Comment (JanuszL) [reply qa/setup_dali_extra.sh:5]: Ill change when https://github.com/NVIDIA/DALI_extra/pull/51 is merged. Until then I needed to test this PR somehow.

File: qa/setup_dali_extra.sh [hunk 2]
```diff
@@ -13,6 +13,6 @@ if [ ! -d "$DALI_EXTRA_PATH" ] ; then
 fi
 
 pushd "$DALI_EXTRA_PATH"
-git fetch origin ${DALI_EXTRA_VERSION}
-git checkout ${DALI_EXTRA_VERSION}
+git fetch origin
+git checkout more_video_containers
```
[the remark this answers is not in the document]
Comment (JanuszL) [reply qa/setup_dali_extra.sh:17]: As above

File: dali/operators/reader/video_reader_op_test.cc [hunk 3]
```diff
@@ -257,9 +260,12 @@ TEST_F(VideoReaderTest, Vp9Profile2) {
       "Decoder hardware does not support this video codec"
       " and/or chroma format";
 
+  // richer FFmpeg configuration leads to different behaviour of VFR heuristics so dissable it for
+  // for this video
```
[the remark this answers is not in the document]
Comment (JanuszL) [reply dali/operators/reader/video_reader_op_test.cc]: Done

File: dali/operators/reader/video_reader_op_test.cc [hunk 4]
```diff
@@ -221,9 +221,12 @@ TEST_F(VideoReaderTest, Vp9Profile0) {
   Pipeline pipe(1, 1, 0);
   const int sequence_length = 60;
 
+  // richer FFmpeg configuration leads to different behaviour of VFR heuristics so dissable it for
+  // for this video
```
[the remark this answers is not in the document]
Comment (JanuszL) [reply dali/operators/reader/video_reader_op_test.cc]: Done

**Review by awolant — APPROVED** [review]

**Review by banasraf — APPROVED** [review]

**Comment by JanuszL** [comment]
!buil

**Pull request merged**